### PR TITLE
DTS recursive symlinks

### PR DESCRIPTION
**Pull request by ekazachkova**
Implementation for issue #502 

Added a new option `--symlinks (-sl)` that supports three possible value:
- follow - follow symlinks (default)
- skip - do not follow symlinks
- filter - follow symlinks but check for cyclic links

Also, dts application properties names was changed

### CODE REVIEW

File: pipe-cli/src/model/data_storage_wrapper.py [hunk 1]
```diff
@@ -407,21 +418,34 @@ def leaf_path(source_path):
...
-            def list_items(path, parent, root=False):
+            def list_items(path, parent, symlinks, visited_symlinks, root=False):
                 for item in os.listdir(path):
                     absolute_path = os.path.join(path, item)
+                    if os.path.islink(absolute_path) and symlinks != AllowedSymlinkValues.FOLLOW:
+                        if symlinks == AllowedSymlinkValues.SKIP:
+                            continue
+                        if symlinks == AllowedSymlinkValues.FILTER:
+                            symlink_target = os.readlink(absolute_path)
+                            if absolute_path.startswith(symlink_target):
```
Comment (mzueva) [inline pipe-cli/src/model/data_storage_wrapper.py]: Could you explain, why do we need this condition?

File: pipe-cli/pipe.py [hunk 2]
```diff
@@ -855,13 +855,19 @@ def storage_move_item(source, destination, recursive, force, exclude, include, q
 @click.option('-l', '--file-list', required=False, help="Path to file with file paths that should be copied. This file "
                                                         "should be tub delimited and consist of two columns: "
                                                         "relative path to file and size.")
+@click.option('-sl', '--symlinks', required=False, default="follow",
```
Comment (mzueva) [inline pipe-cli/pipe.py:858]: Can't we use [Choice](https://click.palletsprojects.com/en/7.x/options/#choice-options) to simplify allowed values handling?

**Review by mzueva — CHANGES_REQUESTED** [review]

File: pipe-cli/pipe.py (hunk 2 above)
Comment (ekazachkova) [reply pipe-cli/pipe.py:858]: thanks, added

File: pipe-cli/src/model/data_storage_wrapper.py (hunk 1 above)
Comment (ekazachkova) [reply pipe-cli/src/model/data_storage_wrapper.py]: fixed

**Review by mzueva — APPROVED** [review]

**Pull request merged**